deployment/common/changeset/link_transfer_timelock.go

```diff
@@ -0,0 +1,177 @@
+		return fmt.Errorf("transfers map must have at least one key")
+	}
+
+	// Check transfers config values.
+	for key, transfers := range cfg.Transfers {
+		if len(transfers) == 0 {
+			return fmt.Errorf("transfers for key %d must have at least one LinkTransfer", key)
+		}
+		for _, transfer := range transfers {
+			if transfer.To == (common.Address{}) {
+				return fmt.Errorf("'to' address for transfers  must be set")
+			}
+			if transfer.Value == nil {
+				return fmt.Errorf("value for transfers must be set")
+			}
+			if transfer.Value == big.NewInt(0) {
+				return fmt.Errorf("value for transfers must be non-zero")
+			}
+		}
+	}
+	// Check that Transfers and StartingOpCount have the same keys
+	for key := range cfg.Transfers {
+		_, err := chain_selectors.GetSelectorFamily(key)
+		if err != nil {
+			return fmt.Errorf("invalid chain selector: %w", err)
+		}
+	}
+	if !cfg.UseMCMS {
+		return nil
+	}
+	// Mcms specific configs
+	if cfg.McmsConfig == nil {
+		return fmt.Errorf("mcmsConfig must be set when UseMCMS is true")
+	}
+	// Upper bound for min delay (7 days)
+	if cfg.McmsConfig.MinDelay > 24*7*time.Hour {
+		return fmt.Errorf("minDelay must be less than 7 days")
+	}
+	return nil
+}
+
+// LinkTransferTimelock takes the given link transfers and executes them or creates an MCMS proposal for them.
+func LinkTransferTimelock(e deployment.Environment, req *LinkTransferTimelockConfig) (deployment.ChangesetOutput, error) {
+	err := req.Validate()
+	ctx := e.GetContext()
+	if err != nil {
+		return deployment.ChangesetOutput{}, fmt.Errorf("invalid LinkTransferTimelockConfig: %w", err)
+	}
+	chainSelectors := []uint64{}
+	for chainSelector := range req.Transfers {
+		chainSelectors = append(chainSelectors, chainSelector)
+	}
+	mcmsPerChain := map[uint64]*owner_helpers.ManyChainMultiSig{}
+
+	timelockAddresses := map[mcms.ChainIdentifier]common.Address{}
+	allBatches := []timelock.BatchChainOperation{}
+	for chainSelector := range req.Transfers {
+		chainID := mcms.ChainIdentifier(chainSelector)
+		chain := e.Chains[chainSelector]
+		addrs, err := e.ExistingAddresses.AddressesForChain(chainSelector)
+		linkState, err := MaybeLoadLinkTokenState(chain, addrs)
+		if err != nil {
+			return deployment.ChangesetOutput{}, err
+		}
+		linkAddress := linkState.LinkToken.Address()
+		mcmsState, err := MaybeLoadMCMSWithTimelockState(chain, addrs)
+		if err != nil {
+			return deployment.ChangesetOutput{}, err
+		}
+		timelockAddress := mcmsState.Timelock.Address()
+
+		mcmsPerChain[uint64(chainID)] = mcmsState.ProposerMcm
+
+		timelockAddresses[chainID] = timelockAddress
+		batch := timelock.BatchChainOperation{
+			ChainIdentifier: chainID,
+			Batch:           []mcms.Operation{},
+		}
+		opts := deployment.SimTransactOpts()
+		if !req.UseMCMS {
+			opts = chain.DeployerKey
+		}
+		totalAmount := big.NewInt(0)
+		for _, transfer := range req.Transfers[chainSelector] {
+			tx, err := linkState.LinkToken.Transfer(opts, transfer.To, transfer.Value)
```

Comment:
if the deployer key is used, you need to confirm the transaction

Reply:
added here https://github.com/smartcontractkit/chainlink/pull/15512/commits/f889dfc35991573e31c5a56015b3780ff673a9a5